MovieApp/MovieApp/App/Features/MovieBrowser/Components/SubcategoryView/SubcategoryView+Design.swift

```diff
@@ -0,0 +1,95 @@
+        
+        subcategoriesStack = UIStackView()
+        scrollView.addSubview(subcategoriesStack)
+        
+        addButtons()
+    }
+    
+    func styleViews() {
+        scrollView.contentInset = UIEdgeInsets(top: 0, left: 4 * offset, bottom: 0, right: 0)
+        scrollView.showsHorizontalScrollIndicator = false
+        
+        subcategoriesStack.alignment = .center
+        subcategoriesStack.spacing = 5 * offset
+        subcategoriesStack.axis = .horizontal
+        
+        styleButtons()
+    }
+    
+    func defineLayoutForViews() {
+        scrollView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        
+        subcategoriesStack.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+    }
+    
+}
+
+extension SubcategoryView {
+    
+    func addButtons() {
+        for tag in 0..<maxNumberOfSubcategoryItems {
+            let button = UIButton()
+            button.tag = tag
+            button.addTarget(self, action: #selector(subcategoryButtonPressed), for: .touchUpInside)
+            subcategoriesStack.addArrangedSubview(button)
+        }
+    }
+    
+    func styleButtons() {
+        for button in subcategoriesStack.arrangedSubviews {
+            if let button = button as? UIButton {
+                if(button.tag == selectedSubcategory) {
+                    styleSelectedSubcategory(button)
+                } else {
+                    styleUnselectedSubcategory(button)
+                }
+            }
+        }
+    }
+    
+    func styleSelectedSubcategory(_ button: UIButton) {
+        let font = UIFont(name: Fonts.proximaBold, size: 16) ?? .systemFont(ofSize: 16)
+        button.setAttributedTitle(NSAttributedString(
+                                    string: "Subcategory",
+                                    attributes: [NSAttributedString.Key.font: font,
+                                                 NSAttributedString.Key.underlineStyle: NSUnderlineStyle.thick.rawValue,
+                                                 NSAttributedString.Key.underlineColor: UIColor.black,
+                                                 NSAttributedString.Key.foregroundColor: UIColor.black]),
+                                  for: .normal)
+    }
+    
+    func styleUnselectedSubcategory(_ button: UIButton) {
+        let font = UIFont(name: Fonts.proximaNovaSemiBold, size: 16) ?? .systemFont(ofSize: 16)
+        button.setAttributedTitle(NSAttributedString(
+                                    string: "Subcategory",
+                                    attributes: [NSAttributedString.Key.font: font,
+                                                 NSAttributedString.Key.foregroundColor: UIColor.secondaryGray]),
+                                  for: .normal)
+    }
+    
+    @objc func subcategoryButtonPressed(sender: UIButton) {
+        if let button = subcategoriesStack.arrangedSubviews[selectedSubcategory] as? UIButton {
```

Comment:
This should be `guard`

Reply:
[Removed](https://github.com/lihter/Movie-App/pull/24/files#diff-b2b46513e62a9e31165f1b658855d74d9d487f3eb70e301c3940cebbcbe2e529L87) this part of the code since I've [implemented](https://github.com/lihter/Movie-App/pull/24/files#diff-9d4d246c1c825ccda3fc16275b719b1c8e8532c9fab35275ccf7f4542be791afR8) so the subcategories array is now [LocalSubcategory]?.